src/Symfony/Component/DomCrawler/FormFieldRegistry.php

```diff
@@ -163,13 +155,12 @@ private static function create($base, array $values)
     /**
      * Transforms a PHP array in a list of fully qualified name / value.
      *
-     * @param array  $array  The PHP array
-     * @param string $base   The name of the base field
-     * @param array  $output The initial values
+     * @param array $array  The PHP array
+     * @param array $output The initial values
      *
      * @return array The list of fields as [string] Fully qualified name => (mixed) value)
      */
-    private function walk(array $array, $base = '', array &$output = [])
+    private function walk(array $array, ?string $base = '', array &$output = [])
```

Comment:
Shouldn't `?string` be instead `string` ?

Reply:
is `null` actually allowed ?

Reply:
if we don't allow null, it break lots of test: 
```
TypeError: Argument 2 passed to Symfony\Component\DomCrawler\FormFieldRegistry::walk() must be of the type string, null given, called in /home/hamza/projet/contrib/symfony/src/Symfony/Component/DomCrawler/FormFieldRegistry.php on line 133

```

Reply:
So, it means that we need to  ensure that `$this->base` cannot be `null` but an empty string instead. Or we need to change the calling code.

Reply:
we can set the default value of `$this->base` to an empty string, that's the proposal I've pushed.